### PR TITLE
Upload sccache stats

### PR DESCRIPTION
**Pull request by driazati**
This adds some tracking to metrics.pytorch.org for sccache build stats per environment

### CODE REVIEW

File: tools/stats/upload_sccache_stats.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                    data[name] = parse_value(values[1])
+
+        # The data from sccache is always the same so this should be fine, if it
+        # ever changes we will probably need to break this out so the fields
+        # we want are hardcoded
+        register_rds_schema("sccache_stats", schema_from_sample(data))
+
+        rds_write(
+            "sccache_stats", [data], only_on_master=False
+        )  # TODO: disable this once it's been tested
```
Comment (janeyx99) [inline tools/stats/upload_sccache_stats.py:77]: will this be removed this PR

**Review by zhouzhuojie — APPROVED** [review]

File: tools/stats/upload_sccache_stats.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+def sprint(*args: Any) -> None:
+    print("[sccache_stats]", *args, file=sys.stderr)
+
+
+def parse_value(value: str) -> Any:
+    try:
+        return int(value)
+    except ValueError:
+        if value.endswith(" s"):
```
Comment (janeyx99) [inline tools/stats/upload_sccache_stats.py:21]: could you maybe add some context for this in a comment?

File: tools/stats/upload_sccache_stats.py (hunk 1 above)
Comment (driazati) [reply tools/stats/upload_sccache_stats.py:77]: Yeah before landing, it's just so we can see some [test data](https://metrics.pytorch.org/explore?orgId=1&left=%5B%22now-1h%22,%22now%22,%22Metrics%22,%7B%22datasource%22:%22Metrics%22,%22format%22:%22table%22,%22timeColumn%22:%22updated_at%22,%22metricColumn%22:%22none%22,%22group%22:%5B%5D,%22where%22:%5B%5D,%22select%22:%5B%5B%7B%22type%22:%22column%22,%22params%22:%5B%22average_cache_read_hit%22%5D%7D%5D%5D,%22rawQuery%22:true,%22rawSql%22:%22select%20*%20from%20sccache_stats%20limit%2010;%22,%22table%22:%22sccache_stats%22,%22timeColumnType%22:%22timestamp%22%7D%5D)

File: .github/templates/windows_ci_workflow.yml.j2 [hunk 3]
```diff
@@ -110,10 +112,14 @@ jobs:
...
           .circleci/scripts/windows_cudnn_install.sh
 {%- endif %}
+      !{{ common.parse_ref() }}
       - name: Build
         shell: bash
         env:
           PYTORCH_FINAL_PACKAGE_DIR: /c/${{ github.run_id }}/build-results/
+          CIRCLE_BRANCH: ${{ steps.parse-ref.outputs.branch }}
+          CIRCLE_PR_NUMBER: ${{ github.event.pull_request.number }}
+          CIRCLE_SHA1: ${{ github.event.pull_request.head.sha || github.sha }}
```
Comment (janeyx99) [inline .github/templates/windows_ci_workflow.yml.j2:122]: Should these just be toplevel env at this point?

**Review by janeyx99 — APPROVED** [review]
🎉

File: .github/templates/windows_ci_workflow.yml.j2 (hunk 3 above)
Comment (janeyx99) [reply .github/templates/windows_ci_workflow.yml.j2:122]: This doesn't have to be this PR

Comment (driazati) [reply .github/templates/windows_ci_workflow.yml.j2:122]: agreed, `CIRCLE_BRANCH` relies on an actual step though so we have to keep that one per-step

**Review by malfet — APPROVED** [review]

File: .github/templates/linux_ci_workflow.yml.j2 [hunk 4]
```diff
@@ -133,13 +136,23 @@ jobs:
       - name: Pull Docker image
         run: |
           !{{ common.pull_docker("${DOCKER_IMAGE}") }}
+          docker pull "${DOCKER_IMAGE}"
```
Comment (malfet) [inline .github/templates/linux_ci_workflow.yml.j2:139]: Merge artefact?
```suggestion
```

**Review by malfet — DISMISSED** [review]

File: .github/templates/linux_ci_workflow.yml.j2 (hunk 4 above)
Comment (zhouzhuojie) [reply .github/templates/linux_ci_workflow.yml.j2:139]: good catch!

Comment (malfet) [reply .github/templates/linux_ci_workflow.yml.j2]: To be frank, spotted it only because I just did a very similar conflict resolution

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**